mxcube3/ui/actions/queue.js

```diff
@@ -402,7 +400,7 @@ export function addTask(sampleIDs, parameters, runNow) {
                      checked: true };
 
       if (!state.queue.queue.includes(sampleID)) {
-        const sample = state.sampleGrid.sampleList[sampleID];
+        const sample = Object.assign({}, state.sampleGrid.sampleList[sampleID]);
```

Comment:
we need to check if the sample(s) are already in the queue, and also _push_ for the new task. If we do not copy the object we are actually pushing to the state before anything else happens, and thus, the task will appear in the sample grid before any call to the server is made.